omega-h: add and deploy 9.31.3.dev1

tristan0x opened the pull request.

tristan0x committed:
omega-h: add and deploy 9.31.3.dev1

tristan0x commented:
retest this please Jenkins

pramodk reviewed (dismissed):
Need to check netcdf recipe url?

Commit:
netcdf: add 4.7.3

Commit:
trilinos: now requires netcdf@4.7.3

Commit:
omega-h: disable gmsh parallel format

It seems Omega_h is missing the file FindGmsh.cmake, required when
this feature is enabled.

Commit:
omega-h: bump version to support gmsh parallel format

pramodk commented:
please
